tests/testsuite/main.rs

```diff
@@ -193,7 +193,12 @@ mod weak_dep_features;
 mod workspaces;
 mod yank;
 
-use cargo_test_support::prelude::*;
+use crate::prelude::*;
+
+pub mod prelude {
+    pub use crate::utils::ext::CargoProjectExt;
```

Comment:
Can you split out the creation of `CargoProjectExt` from the prelude, as in create the prelude in one commit and then `CargoProjectExt` after it?

This touches a lot of files and so being minimal with what is changed can make it a lot easier to review or come back later and see what was done

See also https://github.com/rust-lang/cargo/pull/15692#discussion_r2161860056